src/Sentry/PlatformAbstractions/FrameworkInfo.NetFx.cs

```diff
@@ -81,10 +102,15 @@ public static partial class FrameworkInfo
         /// <summary>
         /// Get all .NET Framework installations in this machine
         /// </summary>
+        /// <param name="registryLock">Blocks this function to access the System's registry.</param>
         /// <seealso href="https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#to-find-net-framework-versions-by-querying-the-registry-in-code-net-framework-1-4"/>
         /// <returns>Enumeration of installations</returns>
-        public static IEnumerable<FrameworkInstallation> GetInstallations()
+        public static IEnumerable<FrameworkInstallation> GetInstallations(bool registryLock = false)
         {
+            if (registryLock || RegistryLocked)
+            {
+                yield break;
+            }
```

Comment:
What's the point of passing a parameter that just makes the function exit immediately? Wouldn't it be better to just _not call_ the function instead?